cli/module_graph2.rs

```diff
@@ -1005,45 +1389,17 @@ impl GraphBuilder2 {
     Ok(())
   }
 
-  /// Insert a module into the graph based on a module specifier.  The module
-  /// and any dependencies will be fetched from the handler.  The module will
-  /// also be treated as a _root_ module in the graph.
-  pub async fn insert(
-    &mut self,
-    specifier: &ModuleSpecifier,
-  ) -> Result<(), AnyError> {
-    self.fetch(specifier)?;
-
-    loop {
-      let cached_module = self.pending.next().await.unwrap()?;
-      self.visit(cached_module)?;
-      if self.pending.is_empty() {
-        break;
-      }
-    }
-
-    if !self.graph.roots.contains(specifier) {
-      self.graph.roots.push(specifier.clone());
-    }
-
-    Ok(())
-  }
-
   /// Move out the graph from the builder to be utilized further.  An optional
   /// lockfile can be provided, where if the sources in the graph do not match
-  /// the expected lockfile, the method with error instead of returning the
-  /// graph.
+  /// the expected lockfile, an error will be logged and the process will exit.
```

Comment:
What's the reasoning behind making `get_graph` infallible? Seem like the `Result` return value that was here before would be a better separation of concerns.

Reply:
A couple reasons... `get_graph()` should be infallible in my opinion... it should externally be a simple "move" out.

Also, my opinion is locking is an internal concern.  It should be terminal and not left up to the consumer of the API to figure that out.  Because the lockfile was asserted, it is meant to be strictly enforced.  Bubbling up a result means that the consumer could ignore a failure and continue, when we know objectively something is wrong with the graph.

Lastly, we exit with code 10.  If we moved that out of here, then there would be about 5-6 places where we would have to downcast and any error to a lock error, to be able to be certain it was a lock error, and then branch to exit the process with an error of 10.

I had originally made in fallible and it was starting to get messy and difficult.